### ISSUE TITLE
Change Request: Simplify usage of TypeScript config files with Node

### ISSUE THREAD

**Issue by LangLangBart** [eslint/eslint#20903]
### ESLint version

10.4.0

### What problem do you want to solve?

For Node **>=22.18.0**, you must pass the `unstable_native_nodejs_ts_config` flag or use `jiti` when using a TypeScript config file.

❌ Fails

```bash
podman run --quiet --rm node:22.18-alpine sh -c '
cd /tmp && npm init -y > /dev/null
npm install eslint > /dev/null
echo "export default [{}]" > eslint.config.mts
echo "foo;" > foo.js
npx eslint && echo OK'

# ...
# Error: The 'jiti' library is required for loading TypeScript configuration files. Make sure to install it.
#  ...
```

✅ Success with `jiti`

```bash
podman run --quiet --rm node:22.18-alpine sh -c '
cd /tmp && npm init -y > /dev/null
npm install eslint jiti > /dev/null
echo "export default [{}]" > eslint.config.mts
echo "foo;" > foo.js
npx eslint && echo OK'

# OK
```

✅ Success with `--flag unstable_native_nodejs_ts_config`

```bash
podman run --quiet --rm node:22.18-alpine sh -c '
cd /tmp && npm init -y > /dev/null
npm install eslint > /dev/null
echo "export default [{}]" > eslint.config.mts
echo "foo;" > foo.js
npx eslint --flag unstable_native_nodejs_ts_config && echo OK'

# OK
```

### What do you think is the correct solution?

1. Drop **Node 20.x** -  it's EOL[^1] and can never support TS stripping
2. Bump minimum **Node 22** to **>=22.18.0**[^2] - the first version where TS stripping is enabled by default
3. Remove all mentions and the need to use `jiti` from the docs
4. Remove the `unstable_native_nodejs_ts_config` flag 

https://github.com/eslint/eslint/blob/9067f9492ec998afc5b4f057a477ecf6ebd45e44/lib/shared/flags.js#L28-L35

Using ` eslint.config.{mts,cts,ts}` just works. 😀

[^1]: https://nodejs.org/en/about/previous-releases
[^2]: https://nodejs.org/en/blog/release/v22.18.0

### Participation

- [ ] I am willing to submit a pull request for this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by kavixo** [comment]
Might be better to reopen #19985 since there’s already more discussion and context there.

**Comment by DMartens** [comment]
There is already a [PR](https://github.com/eslint/eslint/pull/20796) open for updating the documentation.
As we are not going to update the minimum Node.js version until the next major version, I am going to close this.

**Comment by LangLangBart** [comment]
@DMartens, the referenced PR is unrelated to this issue; the current documentation already includes all the information. Closing it as a duplicate of 19985 would have been more understandable.

**Comment by LangLangBart** [comment]
Can one of the members reopen the issue, e.g. @nzakas ? As I don't find any closure from the reason given.

Why is it not important to remove needless friction?

**Comment by DMartens** [comment]
My point was that we are highly unlikely going to bump the major version of Node.js before a new major version of ESLint (bumping the Node.js version would also create friction).
As said in #20290 the problem is that there is [behavioral change](https://github.com/eslint/eslint/issues/19985#issuecomment-3160435360) and as that issue was closed last month I do not think we will unflag TS configs soon.
Further can you please specify by friction?
I tried using `eslint.config.ts` and did not need any flags as `npm` as well as `pnpm` also installed jiti. 

I chose #19985 as I understood it as a documentation problem but this is also as already mentioned a duplicate of #20290 but I can choose only one (but yes #20290 matches better).

**Comment by LangLangBart** [comment]
> I tried using `eslint.config.ts` and did not need any flags as `npm` as well as `pnpm` also installed jiti.

The isolated Podman/Docker environment in the issue description above tells otherwise.

`jiti` is a **devDependency** - one reason was no overhead for js users
(#18134); the user always has to opt in (friction)

- Ref: https://npmgraph.js.org/?q=eslint

In #19401 they added native TypeScript Support, but have hidden it behind a flag (friction)

Users keep bumping into the jiti error:

- https://github.com/eslint/eslint/pull/19401#issuecomment-3677874106
- https://github.com/eslint/eslint/issues/19357#issuecomment-4006468770
- 🙋‍♂️me
- ...

**Comment by kirkwaiblinger** [comment]
@DMartens Is there a reason this needs to be closed though? Why not keep this (or https://github.com/eslint/eslint/issues/19985) open as a change request for the next major? 

Other issues that are breaking are open, such as https://github.com/eslint/eslint/issues/20620, https://github.com/eslint/eslint/issues/20776, https://github.com/eslint/eslint/issues/20689...

**Comment by DMartens** [comment]
This is a good point (I closed it initially because I thought of it as a duplicate).

**Comment by lumirlumir** [comment]
I’m actually in favor of this change for the next major version, as it can use Node’s native feature as the primary way to support `.ts` config files, without any external dependencies or flags.

Although there are some behavioral differences between `jiti` and native type stripping, many other tooling packages are trying to use Node’s native TypeScript support as the primary way to support `.ts` config files, such as [`prettier`](https://prettier.io/docs/configuration#typescript-configuration-files), [`webpack`](https://webpack.js.org/guides/typescript/?utm_source=chatgpt.com#ways-to-use-typescript-in-webpackconfigts), [`next`](https://nextjs.org/docs/app/api-reference/config/typescript#using-nodejs-native-typescript-resolver-for-nextconfigts), and others.

I can work on this issue when the time comes.

**Comment by bakkot** [comment]
I don't understand why this is regarded as a major change. Specifically, it would be straightforward to use `jiti` for `.ts` files if it is installed, and only fall back to native type stripping if it is not installed. This would only have the effect of making an error into a non-error. A future major could drop support for `jiti`.

[The only reason I've heard not to do this](https://github.com/eslint/eslint/issues/19985#issuecomment-3160435360) is "it would be too confusing of an end user experience to have the same version of ESLint have different behaviors depending on which version of Node.js you're using". I don't think this is a big deal, especially given that all the versions where this wouldn't work are EOL, and especially as compared to the current confusion involved in loading `.ts` config files. 

Can someone explain the motivation for not just doing this now?

**Comment by GrantGryczan** [comment]
I agree with the previous comment, and furthermore, I personally feel that my end user experience is _more_ confusing because of this. I have a modern version of Node, so it makes no sense that ESLint artificially blocks me from using `eslint.config.ts` by requiring a useless dependency.

I don't buy that this is a breaking change. The fix is simply to remove an artificial block, which doesn't make any existing setup fail.
